django/contrib/staticfiles/storage.py

```diff
@@ -296,7 +295,7 @@ def path_level(name):
                     if hashed_file_exists:
                         self.delete(hashed_name)
                     # then save the processed result
-                    content_file = ContentFile(force_bytes(content))
+                    content_file = ContentFile(content.encode())
```

Comment:
I'm not sure if this should be `.encode(settings.FILE_CHARSET)` to preserve the file encoding. The file is reading using this encoding on [line 286](https://github.com/django/django/blob/495abe00951ceb9787d7f36590f71aa14c973d3d/django/contrib/staticfiles/storage.py#L287). The previous use of `force_bytes()` didn't do this, so I didn't change behavior. Thoughts?

Reply:
Seems reasonable at a quick glance, can you add a test to demonstrate the bug?

Reply:
I have reconsidered. I no longer believe the file should be written with the `settings.FILE_CHARSET` encoding. The [documentation states](https://docs.djangoproject.com/en/dev/ref/settings/#file-charset):

> The character encoding used to **decode any files read from disk**. This includes template files and initial SQL data files.

Emphasis mine.

The documentation suggests the setting is used only for reading files. As this operation is writing a file, the `settings.FILE_CHARSET` need not be involved. This same reasoning has been used in at lease [one past discussion](https://code.djangoproject.com/ticket/21762#comment:3). From @claudep:

> `settings.FILE_CHARSET` is documented and used in Django for reading files, not for writing files.

So, perhaps we should continue this tradition and continue writing the file encoded as UTF-8. WDYT?

---

As an aside, perhaps `settings.FILE_CHARSET` is no longer necessary and could be considered for deprecation. Its use is extremely limited in the code. We could instead document that files should be UTF-8 encoded. If there is any interest, I'm happy to do the work to make it happen. Its existence is [justified as](https://docs.djangoproject.com/en/dev/ref/unicode/#templates):

> But the common case is to read templates from the filesystem, and this creates a slight complication: not all filesystems store their data encoded as UTF-8. If your template files are not stored with a UTF-8 encoding, set the FILE_CHARSET setting to the encoding of the files on disk. When Django reads in a template file, it will convert the data from this encoding to Unicode. 

Which seems quite suspect to me. I'm not aware of any modern environment that can't UTF-8 encode files to disk.

Reply:
I've solicited input: https://groups.google.com/d/topic/django-developers/7bZbYVV6hg4/discussion